gorp.go

```diff
@@ -239,43 +239,63 @@ func columnToFieldIndex(m *DbMap, t reflect.Type, name string, cols []string) ([
 		tableMapped = true
 	}
 
+	lowerCaseCols := make([]string, len(cols))
+	for i, v := range cols {
+		lowerCaseCols[i] = strings.ToLower(v)
+	}
+
 	// Loop over column names and find field in i to bind to
 	// based on column name. all returned columns must match
 	// a field in the i struct
-	missingColNames := []string{}
-	for x := range cols {
-		colName := strings.ToLower(cols[x])
-		field, found := t.FieldByNameFunc(func(fieldName string) bool {
-			field, _ := t.FieldByName(fieldName)
-			cArguments := strings.Split(field.Tag.Get("db"), ",")
-			fieldName = cArguments[0]
-
-			if fieldName == "-" {
-				return false
-			} else if fieldName == "" {
-				fieldName = field.Name
-			}
-			if tableMapped {
-				colMap := colMapOrNil(table, fieldName)
-				if colMap != nil {
-					fieldName = colMap.ColumnName
-				}
+	var foundColCount int
+	// better to use recursive function to loop
+	t.FieldByNameFunc(func(name string) bool {
+		field, _ := t.FieldByName(name)
+		cArguments := strings.Split(field.Tag.Get("db"), ",")
+		fieldName := cArguments[0]
+
+		if fieldName == "-" {
+			return false
+		} else if fieldName == "" {
+			fieldName = field.Name
+		}
+
+		if tableMapped {
+			colMap := colMapOrNil(table, fieldName)
+			if colMap != nil {
+				fieldName = colMap.ColumnName
 			}
-			return colName == strings.ToLower(fieldName)
-		})
-		if found {
-			colToFieldIndex[x] = field.Index
 		}
-		if colToFieldIndex[x] == nil {
-			missingColNames = append(missingColNames, colName)
+
+		fieldName = strings.ToLower(fieldName)
+
+		for i, v := range lowerCaseCols {
+			if v != fieldName {
+				continue
+			}
+			foundColCount++
+			colToFieldIndex[i] = field.Index
+		}
```

Comment:
Since the size of lowerCaseCols expected to be small (one must be really tricky to have a struct with > 100 fields) so I think this not the case.